core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -73,47 +138,68 @@ object JsonAST {
       case _ => Nil
     }
 
-    /** XPath-like expression to query JSON fields by name. Returns all matching fields.
-      * <p>
-      * Example:<pre>
-      * json \\ "name"
-      * </pre>
-      */
+    /**
+     * Find all children of a `JObject` with the matching name, returning an empty `JObject` is no matches are found.
+     *
+     * Example:
+     *
+     * {{{
+     * json \\ "name"
+     * }}}
+     */
     def \\(nameToFind: String): JValue = {
       def find(json: JValue): List[JField] = json match {
-        case JObject(l) => l.foldLeft(List[JField]()) { 
-          case (a, JField(name, value)) => 
-            if (name == nameToFind) {
-              a ::: List(JField(name, value)) ::: find(value)
-            } else {
-              a ::: find(value)
-            }
-        }
-        case JArray(l) => l.foldLeft(List[JField]())((a, json) => a ::: find(json))
-        case _ => Nil
+        case JObject(fields) =>
+          fields.foldLeft(List[JField]()) {
+            case (matchingFields, JField(name, value)) =>
+              matchingFields :::
+              List(JField(name, value)).filter(_.name == nameToFind) :::
+              find(value)
+          }
+
+        case JArray(fields) =>
+          fields.foldLeft(List[JField]()) { (matchingFields, children) =>
+            matchingFields ::: find(children)
+          }
+
+        case _ =>
+          Nil
       }
+
       find(this) match {
         case JField(_, x) :: Nil => x
         case xs => JObject(xs)
       }
     }
 
-    /** XPath-like expression to query JSON fields by type. Matches only fields on
-      * next level.
-      * <p>
-      * Example:<pre>
-      * json \ classOf[JInt]
-      * </pre>
-      */
+    /** 
+     * Find immediate children of this `JValue` that match the specific type.
+     *
+     * This is only particularly useful if you have a `JObject` and are interested in finding all children that are,
+     * say, integers. YOu might do so like this:
+     *
+     * {{{
+     * json \ classOf[JInt]
+     * }}}
+     *
+     * This method does require that whatever type you're searching for is subtype of `JValue`.
+     */
     def \[A <: JValue](clazz: Class[A]): List[A#Values] =
       findDirect(children, typePredicate(clazz) _).asInstanceOf[List[A]] map { _.values }
 
-    /** XPath-like expression to query JSON fields by type. Returns all matching fields.
-      * <p>
-      * Example:<pre>
-      * json \\ classOf[JInt]
-      * </pre>
-      */
+    /**
+     * Find all descendents of this `JValue` that match the specific type.
+     *
+     * This is only particularly useful if you have a `JObject` and are interested in finding all descendents that are
+     * of a particular type. This method functions a lot like its single-backslash cousin, excepting that it will
+     * dive deep into the `JObject` to find matches.
+     *
+     * You might use this as follows:
+     *
+     * {{{
+     * json \\ classOf[JInt]
```

Comment:
Again re: example—I actually really like this style of describing the analogue between `\` and `\\`, and I think it may be the right way to document the `String`-based `\\` above as well.